crates/sncast/src/helpers/devnet_detection.rs

```diff
@@ -0,0 +1,312 @@
+use std::process::Command;
+
+#[derive(Debug, Clone)]
+struct DevnetInfo {
+    host: String,
+    port: u16,
+}
+
+#[derive(Debug)]
+enum FindDevnetError {
+    None,
+    Multiple,
+    CommandFailed,
+}
+
+pub fn detect_devnet_url() -> Result<String, String> {
+    detect_devnet_from_processes()
+}
+
+#[must_use]
+pub fn is_devnet_running() -> bool {
+    detect_devnet_from_processes().is_ok()
+}
+
+fn detect_devnet_from_processes() -> Result<String, String> {
+    match find_devnet_process_info() {
+        Ok(info) => Ok(format!("http://{}:{}", info.host, info.port)),
+        Err(FindDevnetError::Multiple) => {
+            Err("Multiple starknet-devnet instances found. Please use --url to specify which one to use.".to_string())
+        }
+        Err(FindDevnetError::None | FindDevnetError::CommandFailed) => {
+            // Fallback to default starknet-devnet URL if reachable
+            if is_port_reachable("127.0.0.1", 5050) {
```

Comment:
Maybe let's move this host and port to consts (e.g. `DEFAULT_DEVNET_HOST` and `DEFAULT_DEVNET_PORT`) as they are used in other places too?